pallets/committee/src/types.rs

```diff
@@ -172,3 +175,32 @@ pub enum Vote {
     Nay,
     Abstain,
 }
+
+/// An implementation of EnsureOrigin
+///
+//  This is for the extrinsics only can be called after the
+/// approval of the committee
+pub struct EnsureApprovedByCommittee<N: U32, AccountId, BlockNumber>(
+    sp_std::marker::PhantomData<(N, AccountId, BlockNumber)>,
+);
+impl<
+        O: Into<Result<CommitteeOrigin<AccountId, BlockNumber>, O>>
+            + From<CommitteeOrigin<AccountId, BlockNumber>>,
+        N: U32,
+        AccountId: PartialEq + Default,
+        BlockNumber: Default,
+    > EnsureOrigin<O> for EnsureApprovedByCommittee<N, AccountId, BlockNumber>
+{
+    type Success = ();
```

Comment:
Could make `type Success = AccountId` and then return the AccountId on ok. This would make it more consistent with the existing `ensure_signed`.

Reply:
> Because the proposal is either accepted or rejected, don't think we consider cases were the vote count matters if a simple majority approved of the proposal?

This `U32` is for `min_votes`, https://github.com/ChainSafe/PINT/blob/604ff955fc3b282cba5801c172f9ff6af61aaf2d/pallets/committee/src/types.rs#L157-L160

Reply:
Yes, but that's already checked when the proposal is closed/executed 
https://github.com/ChainSafe/PINT/blob/604ff955fc3b282cba5801c172f9ff6af61aaf2d/pallets/committee/src/lib.rs#L361-L363